promising/promise.py

```
@@ -631,6 +680,26 @@ def _ensure_from_loop_full_unpacking_scheduled(self) -> None:
+        is deferred to the event loop via ``call_soon_threadsafe``. Any
+        exception raised from that callback would otherwise propagate to the
+        loop's default exception handler and leave the Promise stuck in a
+        non-terminal state. This wrapper instead routes the exception through
+        ``_force_internal_error_finish_from_loop`` so the Promise is settled
+        as an internal error.
+
+        NOTE: This method can only be used from the event loop of the Promise.
+        """
+        try:
+            self._ensure_from_loop_full_unpacking_scheduled()
+        except BaseException as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `BaseException`

(BLE001)